akka-actor/src/main/scala/akka/pattern/AskSupport.scala

```diff
@@ -181,6 +183,9 @@ private[akka] final class PromiseActorRef private (val provider: ActorRefProvide
   import AbstractPromiseActorRef.stateOffset
   import AbstractPromiseActorRef.watchedByOffset
 
+  // This is necessary for weaving the PromiseActorRef into the asked message, i.e. the replyTo pattern.
+  @volatile var messageName = "unknown"
```

Comment:
What's the C&E situation here?

"+        val a = PromiseActorRef(ref.provider, timeout, targetName = actorSel, sender)
+        a.messageName = message.getClass.getName"